src/Take.Elephant.Azure/AzureEventHubReceiverQueue.cs

```diff
@@ -1,148 +1,144 @@
         {
-            await (_receiver?.CloseAsync() ?? Task.CompletedTask);
-            await _eventHubClient.CloseAsync();
+            await _consumer.CloseAsync().ConfigureAwait(false);
         }
 
-        private async Task<bool> OpenIfRequiredAsync(CancellationToken cancellationToken)
+        public Task OpenAsync(CancellationToken cancellationToken)
         {
-            if (_receiver != null) return false;
+            return StartConsumerTaskIfNotAsync(cancellationToken);
+        }
 
-            await _openSemaphore.WaitAsync(cancellationToken);
+        public virtual async ValueTask DisposeAsync()
+        {
+            _cts.Dispose();
+            _consumerStartSemaphore.Dispose();
+            await _consumer.DisposeAsync();
+        }
+
+        public event EventHandler<ExceptionEventArgs> ConsumerFailed;
+
+        private async Task StartConsumerTaskIfNotAsync(CancellationToken cancellationToken)
+        {
+            if (_consumerTask != null) return;
+
+            await _consumerStartSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
             try
             {
-                if (_receiver != null) return false;
-                _receiver = _eventHubClient.CreateReceiver(_consumerGroupName, _partitionId, _eventPosition, _receiverOptions);
-                return true;
+                if (_consumerTask == null)
+                {
+                    _consumerTask = Task
+                        .Factory
+                        .StartNew(
+                            () => ConsumeAsync(_cts.Token),
+                            TaskCreationOptions.LongRunning)
+                        .Unwrap();
+                }
             }
             finally
             {
-                _openSemaphore.Release();
+                _consumerStartSemaphore.Release();
             }
         }
 
-        private T CreateItem(EventData eventData)
-        {
-            if (eventData == null) return default;
-            return _serializer.Deserialize(Encoding.UTF8.GetString(eventData.Body.Array));            
-        }
-
-        internal class ExponentialInterval
+        private async Task ConsumeAsync(CancellationToken cancellationToken)
         {
-            private readonly TimeSpan _initialInterval;
-            private readonly TimeSpan _maxInterval;
-
-            public ExponentialInterval(TimeSpan initialInterval, TimeSpan maxInterval)
-            {
-                _initialInterval = initialInterval;
-                _maxInterval = maxInterval;
-            }
-            
-            public int Count { get; private set;  }
-            
-            public TimeSpan Interval
+            while (!cancellationToken.IsCancellationRequested)
             {
-                get
+                try
+                {
+                    await foreach (var value in _consumer.ReadEventsAsync(cancellationToken).ConfigureAwait(false))
```

Comment:
By the [docs](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.consumer.eventhubconsumerclient.readeventsasync?view=azure-dotnet), `ReadEventAsync` is not recommended for production use :(
The same doc states that `EventProcessorClient` should be used instead, but - as far as I could research - it needs a storage account in order to work...